### PR TITLE
avoid memory allocation in base64decode

### PR DESCRIPTION
**Pull request by neumannt**
Every single call to base64decode allocates a mapping table for all
base64 characters. This is quite wasteful, as the map is in fact static.
We could use a static variable here, but that would have unpleasant consequences
if we ever encounter input with non-valid base64 characters (which
implicitly modifies the map).

The number of character ranges for base64 is quite low (3, plus 4 exceptions),
thus we can simply check that explicitly in code instead of using a dynamic hash table.

### CODE REVIEW

File: include/crow/utility.h
Comment (The-EDev) [inline include/crow/utility.h:547]: Since there wasn't a version released with this function, a breaking change shouldn't be an issue (with the only exception being [this line](https://github.com/CrowCpp/Crow/blob/8f4830b5a14e5ebc2ae9fe70c7d0a3dbc7f4763a/docs/guides/auth.md?plain=1#L32) needing to be changed).

Comment (The-EDev) [inline include/crow/utility.h:547]: Since there wasn't a version released with this function, a breaking change shouldn't be an issue (with the only exception being [this line](https://github.com/CrowCpp/Crow/blob/8f4830b5a14e5ebc2ae9fe70c7d0a3dbc7f4763a/docs/guides/auth.md?plain=1#L32) needing to be changed).

File: include/crow/utility.h
Comment (The-EDev) [inline include/crow/utility.h:553]: would it be best to replace `c - 'a' + 26` with `c - 71` and `c - '0' + 52` with `c + 4` (possibly with comments explaining what the numbers mean)?

File: include/crow/utility.h
Comment (neumannt) [reply include/crow/utility.h]: I have removed that parameter now. Allow that way I noticed that the the std::string interface passed the string by value instead of by reference, I fixed that, too.

File: include/crow/utility.h
Comment (luca-schlecker) [reply include/crow/utility.h:551]: I personally think `(c - 'a') + 26` is quite understandable, each number has a meaning, but with `c - 71` the `71` needs explicit explanation. Could be though, that that's just because I already know what they are meant to do.

Comment (neumannt) [reply include/crow/utility.h:551]: The generated code is [exactly the same](https://godbolt.org/z/Yzcforh6P) the compiler will fold - 'a' + 26 into one constant. Thus I would prefer c - 'a' + 26 over c - 71, as it shows why we perform the subtraction. If you insist I can change that to combined constants, of course.

**Review by luca-schlecker — APPROVED** [review]
Looking good to me, waiting for @The-EDev's final review though.

File: include/crow/utility.h
Comment (The-EDev) [reply include/crow/utility.h:551]: Thanks for clarifying @neumannt

**Review by The-EDev — APPROVED** [review]

**Comment by The-EDev** [comment]
Looks all good, Thanks for your help @neumannt

**Pull request merged**
